### DISCUSSION TITLE
http-server serving different port than assigned, and Chrome not opening

### DISCUSSION THREAD

**Discussion by GonzalesMatthew** [nss-evening-cohort-14/evening-client-side#18] [Help Me!] [answered]
Also, I'm running iOS

Matthew Gonzales

I found this existing ticket, but I seem to have the opposite problem. I think I want my assigned 0.0.0.0:8080 to run and NOT 127.0.0.1, correct?
https://github.com/nss-evening-cohort-14/evening-client-side/discussions/8

I didn't have any error. I just get the message in my VSCode terminal 
"Starting up http-server, serving ./ on: http://0.0.0.0:8080"
and then my Safari browser opens under port 127.0.0.1 instead, and my Chrome browser doesn't open with my project like it did yesterday. 

What have you googled? (Paste links to the things you have found from your google search - You must include at least 3 links)

A possible, but technical solution, and not sure what it's talking about at the moment:
https://stackoverflow.com/questions/33524826/localhost-not-working-in-chrome-127-0-0-1-does-work

History on port 8080 (trying to understand why this port is so important):
https://www.grc.com/port_8080.htm#:~:text=Therefore%2C%20when%20non%2Dadministrators%20wished,secondary%20or%20alternate%20web%20server.

Clearing Chrome cache and cookies:
https://www.google.com/search?q=how+to+clear+cache+and+cookies+chrome&oq=how+to+clear+&aqs=chrome.1.69i57j35i39j0i20i263i457j0i395i433j0i395l6.4486j1j4&sourceid=chrome&ie=UTF-8

What have you tried? Tell us everything you have tried.
I tried clearing my Chrome cache and cookies. Didn't fix it.
I have a work around where I just drag and drop my .html file from VSCode into Chrome and that works, but would like to have an actual solution since I didn't have this issue yesterday.
I'll try going into the more technical answer provided from the first link when I have more time to research what it's referring to exactly, and will post what I find on that if someone doesn't already have an answer for this by then.

**Comment by drteresavasquez** [comment, marked as answer]
So, is the issue that it is defaulting to open in safari, but the command is working? If so, make sure that you have set your default browser to chrome: https://support.apple.com/en-us/HT201607

If that does not work, you may manually have to open the chrome browser as the server is running and type `http://127.0.0.1:8080` in the address bar, so your process would be different. That is not a problem, just a small inconvenience and something that we cannot "fix"

One other option is to run `hs` in your terminal to start your server and then set up an alias to open the browser in a **NEW** terminal window.

You can add this to your ~/.zshrc file:
`alias open="/usr/bin/open -a '/Applications/Google Chrome.app' 'http://127.0.0.1:8080'"`

With this in your zsh file, you can then simply type `open` and chrome will open at port 8080.

**Comment by drteresavasquez** [comment]
@GonzalesMatthew have you been able to try the things recommended? Or, have you found a solution? If so, please let us know how you resolved your issue.

**Reply by GonzalesMatthew** [reply]
@drteresavasquez wasn't able to try anything until now (study group, then work, then food, now here)
manually typing the server into Chrome works best for me. 
added the alias to VSCode but command not recognized for some reason

**Reply by drteresavasquez** [reply]
@GonzalesMatthew You added the alias to the zshrc file not VSCode, right? (Not sure how you would add that) Once you update the zshrc file you have to source that file or close all terminals and start them over for the change to take effect. After that, you should just be able to type `open` in the terminal and the browser will open.
